### ISSUE TITLE
Release Physics-IQ on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [google-deepmind/physics-IQ-benchmark#1]
Hello @rgeirhos 🤗

I'm Niels and work as part of the open-source team at Hugging Face. I discovered your work through Hugging Face's daily papers as yours got featured: https://huggingface.co/papers/2501.09038.
The paper page lets people discuss about your paper and lets them find artifacts about it (your dataset for instance),
you can also claim the paper as yours which will show up on your public profile at HF.

Would you like to host the datasets you've released on https://huggingface.co/datasets?
I see you're using Google Cloud Storage for it. Hosting on Hugging Face will give you more visibility/enable better discoverability, and will also allow people to do:

```python
from datasets import load_dataset

dataset = load_dataset("your-hf-org-or-username/your-dataset")
```

If you're down, leaving a guide here: https://huggingface.co/docs/datasets/loading.
We also support Webdataset, useful for image/video datasets: https://huggingface.co/docs/datasets/en/loading#webdataset.

Besides that, there's the [dataset viewer](https://huggingface.co/docs/hub/en/datasets-viewer) which allows people to quickly explore the first few rows of the data in the browser.

After uploaded, we can also link the datasets to the paper page (read [here](https://huggingface.co/docs/hub/en/model-cards#linking-a-paper)) so people can discover your work.

Let me know if you're interested/need any guidance.

Kind regards,

Niels

**Comment by rgeirhos** [comment]
Hi Niels,
Thanks so much for reaching out! We're interested in potentially hosting the dataset on Hugging Face, but I have a (perhaps unusual) question: given that our dataset is intended as a test dataset, we'd like to avoid it being scraped and ending up in a video model's training dataset (by accident or intentionally). Does Hugging Face provide any guardrails that would help in this regard (e.g. requiring a captcha / password before a dataset can be accessed / ...)?
Cheers,
Robert

**Comment by NielsRogge** [comment]
Hi @rgeirhos thanks for your reply.

Yes in that case I would recommend gated access: https://huggingface.co/docs/hub/en/datasets-gated. It's similar to how models like LLaMa/Mistral are gated. 

There are 2 forms of gated access, either "manual" or "automatic". With "manual", you manually have to review each person who wants to access your dataset. With "automatic", automatic access is given when submitting the contact information, which you can customize.

**Comment by NielsRogge** [comment]
Hi @rgeirhos following up on this, would it be possible to make the dataset available on 🤗 ?

**Comment by sam-motamed** [comment]
Hey @NielsRogge, thanks for following up on this. I'm supposed to be looking into this but have been busy for the past while. I just need to figure out how to enable users to download / load the different FPS options from HuggingFace, Since I've never hosted datasets on HF, I will try to find the answers to my questions in the guide you shared and might reach out to you if questions come up.

**Comment by NielsRogge** [comment]
You might be interested in packaging everything as TAR files using webdataset: https://huggingface.co/docs/hub/datasets-webdataset.

This way, the viewer will work (enabling people to see videos in the browser), and loading with Datasets will be supported: https://huggingface.co/docs/datasets/loading#webdataset.

**Comment by sam-motamed** [comment]
Hi @rgeirhos, I don’t think having the dataset on HF would give a big advantage since we will still need to do our auto FPS adjustment there too, so ```load_dataset``` will have to do that just like our current download pipeline. The only benefit I see is just having another way of downloading the data with HF. If you think it’s worthwhile, we can create a GDM dataset and I’ll push a PR there.

**Comment by rgeirhos** [comment]
Thanks for the comment - I'd say up to you. HF is definitely easier to use for many people to obtain the input, thus might provide more ease of use, but it'd be more of a convenience improvement than a game changer. (If HF, I think we should go for the gated option with automatic access to avoid accidentally landing in someone's training data via scraping)

**Comment by NielsRogge** [comment]
For gated access, see this guide: https://huggingface.co/docs/hub/datasets-gated. 

Happy to help if needed!

**Comment by tuliti** [comment]
@NielsRogge  Great input. We have published the new version, [Physics-IQ Verified](https://huggingface.co/datasets/Anates-Labs-Research/Physics-IQ-Verified), on HF with the recommended gated access (linked in the ReadMe in this repo as well).

**Comment by NielsRogge** [comment]
Awesome! Thanks for linking it
